File: frontend/demo/component/card/card-header-prefix.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/avatar';
import '@vaadin/card';
import { css, html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('card-header-prefix')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-card>
        <!-- tag::[] -->
        <vaadin-avatar slot="header-prefix" name="Lapland"></vaadin-avatar>
        <!-- end::[] -->
        <div slot="title">Lapland</div>
        <div slot="subtitle">The Exotic North</div>
        <div>Lapland is the northern-most region of Finland and an active outdoor destination.</div>
      </vaadin-card>
      <!-- end::snippet[] -->
    `;
  }

  static styles = css`
    vaadin-card {
      max-width: 300px;
    }
  `;
}

File: src/main/java/com/vaadin/demo/component/card/CardHeaderPrefix.java
package com.vaadin.demo.component.card;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.avatar.Avatar;
import com.vaadin.flow.component.card.Card;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;

@Route("card-header-prefix")
public class CardHeaderPrefix extends Div {
    public CardHeaderPrefix() {
        // tag::snippet[]
        Card card = new Card();

        card.setHeaderPrefix(new Avatar("Lapland"));

        card.setTitle(new Div("Lapland"));
        card.setSubtitle(new Div("The Exotic North"));

        card.add(
                "Lapland is the northern-most region of Finland and an active outdoor destination.");
        // end::snippet[]

        card.setMaxWidth("300px");
        add(card);
    }

    public static class Exporter extends DemoExporter<CardHeaderPrefix> { // hidden-source-line
    } // hidden-source-line
}
